### ISSUE TITLE
Cannot import GPy (Debian 8.4, Python 3.4.2)

### ISSUE THREAD

**Issue by ktpolanski** [SheffieldML/GPy#373]
Traceback (most recent call last):
  File "**main**.py", line 12, in <module>
    import gradienttool as gt
  File "/gradienttool/gradienttool.py", line 11, in <module>
    import GPy
  File "/usr/local/lib/python3.4/dist-packages/GPy/**init**.py", line 6, in <module>
    from . import core
  File "/usr/local/lib/python3.4/dist-packages/GPy/core/**init**.py", line 8, in <module>
    from .gp import GP
  File "/usr/local/lib/python3.4/dist-packages/GPy/core/gp.py", line 5, in <module>
    from .. import kern
  File "/usr/local/lib/python3.4/dist-packages/GPy/kern/**init**.py", line 8, in <module>
    from .src.kern import Kern
  File "/usr/local/lib/python3.4/dist-packages/GPy/kern/src/**init**.py", line 1, in <module>
    from . import psi_comp
  File "/usr/local/lib/python3.4/dist-packages/GPy/kern/src/psi_comp/**init**.py", line 20, in <module>
    from .gaussherm import PSICOMP_GH
  File "/usr/local/lib/python3.4/dist-packages/GPy/kern/src/psi_comp/gaussherm.py", line 12, in <module>
    from ....util.linalg import tdot
  File "/usr/local/lib/python3.4/dist-packages/GPy/util/**init**.py", line 9, in <module>
    from . import datasets
  File "/usr/local/lib/python3.4/dist-packages/GPy/util/datasets.py", line 57, in <module>
    json_data=open(path).read()
  File "/usr/lib/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 7276: ordinal not in range(128)

This is quite similar to what a guy was encountering before on that mybinder thing, except I'm not in mybinder. I'm just in a plain Debian, having apt-getted numpy and then pip installed GPy itself (which grabbed scipy along the way). Any idea what's going on here?

**Comment by ken-okabe** [comment]
Hi,

You mention https://github.com/SheffieldML/GPy/issues/319 , and I had the same issue. Eventually I had to work-around by modifying source-code, and my pull request has been already merged. https://github.com/SheffieldML/GPy/pull/378

Since it's not released yet, re-installation via `pip` won't work, but so far, you can modify your local installed code manually;
in your case:
`/usr/local/lib/python3.4/dist-packages/GPy/util/datasets.py`
https://github.com/SheffieldML/GPy/pull/378/commits/025862ec3162c38092b7b518176b59c2c9c4f1e4

**Comment by mzwiessele** [comment]
The patch will come out in due course next week.

**Comment by ken-okabe** [comment]
Thanks a lot for this project, Max!

**Comment by ktpolanski** [comment]
In case you actually rolled out the patch, I'm still getting the error.

**Comment by kayhan-batmanghelich** [comment]
I have similar issue. When I try to `import GPy` in the notebook i get this:

```
An exception has occurred, use %tb to see the full traceback.

SystemExit: This program needs access to the screen.
Please run with a Framework build of python, and only when you are
logged in on the main display of your Mac.
```

the notebook is here:

https://github.com/SheffieldML/notebook/blob/master/GPy/basic_kernels.ipynb

I am using it on MaxOS (El Capitan), with anaconda python:

```
$ conda list
# packages in environment at /Users/kayhan/anaconda:
#
accelerate                2.2.0                np19py27_0    defaults
accelerate_cudalib        2.0                           0    defaults
alabaster                 0.7.7                    py27_0    defaults
anaconda-client           1.4.0                    py27_0    defaults
anaconda                  custom                   py27_0    defaults
anaconda-navigator        1.1.1                    py27_0    defaults
appnope                   0.1.0                    py27_0    defaults
appscript                 1.0.1                    py27_0    defaults
argcomplete               1.0.0                    py27_1    defaults
astropy                   1.0.4                np19py27_0    defaults
autoregressive            0.1.0                     <pip>
babel                     2.2.0                    py27_0    defaults
backports-abc             0.4                       <pip>
backports.ssl-match-hostname 3.4.0.2                   <pip>
backports_abc             0.4                      py27_0    defaults
beautifulsoup4            4.4.1                    py27_0    defaults
bitarray                  0.8.1                    py27_0    defaults
blaze                     0.9.1                    py27_0    defaults
bokeh                     0.11.1                   py27_0    defaults
boost-vpython             1.55.0                        0    https://conda.binstar.org/mwcraig/osx-64/boost-vpython-1.55.0-0.tar.bz2
boto                      2.39.0                   py27_0    defaults
bottleneck                1.0.0                np19py27_0    defaults
brewer2mpl                1.4.1                    py27_0    Chroxvi
cdecimal                  2.3                      py27_0    defaults
cffi                      1.5.2                    py27_0    defaults
chest                     0.2.3                    py27_0    defaults
cloudpickle               0.1.1                    py27_0    defaults
clyent                    1.2.1                    py27_0    defaults
colorama                  0.3.7                    py27_0    defaults
conda                     4.0.6                    py27_0    defaults
conda-build               1.20.0                   py27_0    defaults
conda-env                 2.4.5                    py27_0    defaults
conda-manager             0.4.0                    py27_0    defaults
configobj                 5.0.6                    py27_0    defaults
cryptography              1.3                      py27_0    defaults
cudatoolkit               7.5                           0    defaults
curl                      7.45.0                        0    defaults
cycler                    0.10.0                   py27_0    defaults
cython                    0.23.4                   py27_1    defaults
cytoolz                   0.7.5                    py27_0    defaults
dask                      0.8.1                    py27_0    defaults
datashape                 0.5.1                    py27_0    defaults
decorator                 4.0.9                    py27_0    defaults
dill                      0.2.4                    py27_0    defaults
distribute                0.6.45                   py27_1    defaults
docutils                  0.12                     py27_0    defaults
dynd                      0.7.3.dev1                <pip>
dynd-python               0.7.2                    py27_0    defaults
enum34                    1.1.2                    py27_0    defaults
et-xmlfile                1.0.1                     <pip>
et_xmlfile                1.0.1                    py27_0    defaults
fastcache                 1.0.2                    py27_0    defaults
flask                     0.10.1                   py27_1    defaults
flask-cors                2.1.2                    py27_0    defaults
fonttools                 2.4                  np19py27_0    https://conda.binstar.org/mwcraig/osx-64/fonttools-2.4-np19py27_0.tar.bz2
freetype                  2.5.5                         0    defaults
funcsigs                  0.4                      py27_0    defaults
future                    0.15.2                    <pip>
futures                   3.0.3                    py27_0    defaults
gevent                    1.1.0                    py27_0    defaults
glueviz                   0.7.1                    py27_0    defaults
gpy                       1.0.9                     <pip>
greenlet                  0.4.9                    py27_0    defaults
grin                      1.2.1                    py27_1    defaults
gslrandom                 0.1.1                     <pip>
h5py                      2.5.0                np19py27_3    defaults
hdf5                      1.8.15.1                      2    defaults
heapdict                  1.0.0                    py27_0    defaults
hips-lib                  0.1                       <pip>
idna                      2.0                      py27_0    defaults
iopro                     1.7.2               np19py27_p0    defaults
ipaddress                 1.0.14                   py27_0    defaults
ipykernel                 4.3.1                    py27_0    defaults
ipython                   4.1.2                    py27_1    defaults
ipython-genutils          0.1.0                     <pip>
ipython_genutils          0.1.0                    py27_0    defaults
ipywidgets                4.1.1                    py27_0    defaults
itsdangerous              0.24                     py27_0    defaults
jbig                      2.1                           0    defaults
jdcal                     1.2                      py27_0    defaults
jedi                      0.9.0                    py27_0    defaults
jinja2                    2.8                      py27_0    defaults
jpeg                      8d                            1    defaults
jsonschema                2.4.0                    py27_0    defaults
jupyter                   1.0.0                    py27_2    defaults
jupyter-client            4.2.2                     <pip>
jupyter-console           4.1.1                     <pip>
jupyter-core              4.1.0                     <pip>
jupyter_client            4.2.2                    py27_0    defaults
jupyter_console           4.1.1                    py27_0    defaults
jupyter_core              4.1.0                    py27_0    defaults
libdynd                   0.7.2                         0    defaults
libpng                    1.6.17                        0    defaults
libtiff                   4.0.6                         1    defaults
libxml2                   2.9.2                         0    defaults
libxslt                   1.1.28                        2    defaults
llvmlite                  0.10.0                   py27_0    defaults
locket                    0.2.0                    py27_0    defaults
lxml                      3.6.0                    py27_0    defaults
markupsafe                0.23                     py27_0    defaults
matplotlib                1.4.3                np19py27_3    defaults
mistune                   0.7.2                    py27_1    defaults
mkl                       11.3.1                        0    defaults
mkl-service               1.1.2                    py27_0    defaults
mock                      2.0.0                     <pip>
mpmath                    0.19                     py27_0    defaults
multipledispatch          0.4.8                    py27_0    defaults
nbconvert                 4.1.0                    py27_0    defaults
nbformat                  4.0.1                    py27_0    defaults
networkx                  1.11                     py27_0    defaults
nltk                      3.2                      py27_0    defaults
nose                      1.3.7                    py27_0    defaults
notebook                  4.1.0                    py27_2    defaults
numba                     0.25.0               np19py27_0    defaults
numexpr                   2.4.4                np19py27_0    defaults
numpy                     1.9.3                    py27_1    defaults
odo                       0.4.2                    py27_0    defaults
openpyxl                  2.3.2                    py27_0    defaults
openssl                   1.0.2g                        0    defaults
orange                    2.7.8                    py27_3    defaults
orange-app                1.0                      py27_0    defaults
pandas                    0.16.2               np19py27_0    defaults
paramz                    0.5.5                     <pip>
partd                     0.3.2                    py27_1    defaults
path.py                   8.1.2                    py27_1    defaults
patsy                     0.4.0                np19py27_0    defaults
pbr                       1.9.1                     <pip>
pep8                      1.7.0                    py27_0    defaults
pexpect                   4.0.1                    py27_0    defaults
pgmult (/Users/kayhan/Projects/gitSandBox/pgmult) 0.1                       <pip>
pickleshare               0.5                      py27_0    defaults
pillow                    3.1.1                    py27_0    defaults
pip                       8.1.1                    py27_1    defaults
ply                       3.8                      py27_0    defaults
polygon2                  2.0.6                np19py27_0    https://conda.binstar.org/mwcraig/osx-64/polygon2-2.0.6-np19py27_0.tar.bz2
psutil                    4.1.0                    py27_1    defaults
ptyprocess                0.5                      py27_0    defaults
py                        1.4.31                   py27_0    defaults
pyasn1                    0.1.9                    py27_0    defaults
pyaudio                   0.2.7                    py27_0    defaults
pybasicbayes              0.2.1                     <pip>
pycosat                   0.6.1                    py27_0    defaults
pycparser                 2.14                     py27_0    defaults
pycrypto                  2.6.1                    py27_0    defaults
pycurl                    7.19.5.3                 py27_0    defaults
pyflakes                  1.1.0                    py27_0    defaults
pygments                  2.1.1                    py27_0    defaults
pyhsmm                    0.1.6                     <pip>
pylds                     0.0.1                     <pip>
pyopenssl                 0.15.1                   py27_2    defaults
pyparsing                 2.0.3                    py27_0    defaults
pypolyagamma              0.2                       <pip>
pyqt                      4.11.4                   py27_1    defaults
pytables                  3.2.2                np19py27_0    defaults
pytest                    2.8.5                    py27_0    defaults
python                    2.7.11                        0    defaults
python-dateutil           2.5.1                    py27_0    defaults
python.app                1.2                      py27_4    defaults
pytz                      2016.2                   py27_0    defaults
pyyaml                    3.11                     py27_1    defaults
pyzmq                     15.2.0                   py27_0    defaults
qt                        4.8.7                         1    defaults
qtawesome                 0.3.2                    py27_0    defaults
qtconsole                 4.2.0                    py27_0    defaults
qtpy                      1.0                      py27_0    defaults
readline                  6.2                           2    defaults
redis                     2.6.9                         0    defaults
redis-py                  2.10.3                   py27_0    defaults
requests                  2.9.1                    py27_0    defaults
rope                      0.9.4                    py27_1    defaults
scikit-image              0.11.3               np19py27_0    defaults
scikit-learn              0.16.1               np19py27_0    defaults
scipy                     0.17.0               np19py27_0    defaults
setuptools                20.3                     py27_0    defaults
simplegeneric             0.8.1                    py27_0    defaults
singledispatch            3.4.0.3                  py27_0    defaults
sip                       4.16.9                   py27_0    defaults
six                       1.10.0                   py27_0    defaults
snakeviz                  0.4.0                    py27_0    defaults
snowballstemmer           1.2.1                    py27_0    defaults
sockjs-tornado            1.0.1                    py27_0    defaults
sphinx                    1.3.5                    py27_0    defaults
sphinx-rtd-theme          0.1.9                     <pip>
sphinx_rtd_theme          0.1.9                    py27_0    defaults
spyder                    2.3.8                    py27_1    defaults
sqlalchemy                1.0.12                   py27_0    defaults
sqlite                    3.9.2                         0    defaults
ssl_match_hostname        3.4.0.2                  py27_0    defaults
statsmodels               0.6.1                np19py27_0    defaults
sympy                     1.0                      py27_0    defaults
tables                    3.2.2                     <pip>
terminado                 0.5                      py27_1    defaults
tk                        8.5.18                        0    defaults
toolz                     0.7.4                    py27_0    defaults
tornado                   4.3                      py27_0    defaults
traitlets                 4.2.1                    py27_0    defaults
ttfquery                  1.0.5                    py27_0    https://conda.binstar.org/mwcraig/osx-64/ttfquery-1.0.5-py27_0.tar.bz2
ujson                     1.35                     py27_0    defaults
unicodecsv                0.14.1                   py27_0    defaults
vpnotebook                0.1.3                    py27_0    vpython
vpython                   6.11.0               np19py27_1    https://conda.binstar.org/mwcraig/osx-64/vpython-6.11.0-np19py27_1.tar.bz2
werkzeug                  0.11.4                   py27_0    defaults
wheel                     0.29.0                   py27_0    defaults
wxpython                  3.0                      py27_0    defaults
wxpython-common           3.0.0.0                   <pip>
xlrd                      0.9.4                    py27_0    defaults
xlsxwriter                0.8.4                    py27_0    defaults
xlwings                   0.7.0                    py27_0    defaults
xlwt                      1.0.0                    py27_0    defaults
xz                        5.0.5                         1    defaults
yaml                      0.1.6                         0    defaults
zeromq                    4.1.3                         0    defaults
zlib                      1.2.8                         0    defaults
```

**Comment by mzwiessele** [comment]
Could you update paramz and GPy and try again? We rolled it out : )

**Comment by kayhan-batmanghelich** [comment]
@mzwiessele Thanks for update. In `notenook`, it still has problem:

```
%tb
import GPy 

---------------------------------------------------------------------------
SystemExit                                Traceback (most recent call last)
<ipython-input-1-0109d9ff0d27> in <module>()
----> 1 import GPy

/Users/kayhan/anaconda/lib/python2.7/site-packages/GPy/__init__.pyc in <module>()
     13 from . import testing
     14 from . import kern
---> 15 from . import plotting
     16 
     17 # backwards compatibility

/Users/kayhan/anaconda/lib/python2.7/site-packages/GPy/plotting/__init__.py in <module>()
    121 try:
    122     lib = config.get('plotting', 'library')
--> 123     change_plotting_library(lib)
    124 except NoOptionError:
    125     print("No plotting library was specified in config file. \n{}".format(error_suggestion))

/Users/kayhan/anaconda/lib/python2.7/site-packages/GPy/plotting/__init__.py in change_plotting_library(lib)
     18             import matplotlib
     19             from .matplot_dep.plot_definitions import MatplotlibPlots
---> 20             from .matplot_dep import visualize, mapping_plots, priors_plots, ssgplvm, svig_plots, variational_plots, img_plots
     21             current_lib[0] = MatplotlibPlots()
     22         if lib == 'plotly':

/Users/kayhan/anaconda/lib/python2.7/site-packages/GPy/plotting/matplot_dep/visualize.py in <module>()
      7     import matplotlib as mpl
      8     from mpl_toolkits.mplot3d import Axes3D
----> 9     import visual
     10     visual_available = True
     11 

/Users/kayhan/anaconda/lib/python2.7/site-packages/visual/__init__.py in <module>()
      2 
      3 from visual.visual_all import * # this statement not included in vis/__init__.py
----> 4 from visual_common.create_display import *
      5 
      6 scene = display()

/Users/kayhan/anaconda/lib/python2.7/site-packages/visual_common/create_display.py in <module>()
     26 from visual_common.cvisual import (vector, mag, mag2, norm, cross, rotate,
     27                        comp, proj, diff_angle, set_wait)
---> 28 from visual_common.primitives import (arrow, cylinder, cone, sphere, box, ring, label,
     29                                frame, pyramid, ellipsoid, curve, faces, convex, helix,
     30                                points, text, distant_light, local_light, extrusion,

/Users/kayhan/anaconda/lib/python2.7/site-packages/visual_common/primitives.py in <module>()
     41 
     42 import wx as _wx
---> 43 _App2 = _wx.App()
     44 
     45 _platInfo = _wx.PlatformInformation()

/Users/kayhan/anaconda/lib/python2.7/site-packages/wx-3.0-osx_cocoa/wx/_core.pyc in __init__(self, redirect, filename, useBestVisual, clearSigInt)
   8597                 # TODO: more description is needed for wxMSW...
   8598 
-> 8599             raise SystemExit(msg)
   8600 
   8601         # This has to be done before OnInit

SystemExit: This program needs access to the screen.
Please run with a Framework build of python, and only when you are
logged in on the main display of your Mac.

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-2-2cca71d3b0ad> in <module>()
      1 get_ipython().magic(u'tb')
----> 2 import GPy

/Users/kayhan/anaconda/lib/python2.7/site-packages/GPy/__init__.py in <module>()
      4 warnings.filterwarnings("ignore", category=DeprecationWarning)
      5 
----> 6 from . import core
      7 from . import models
      8 from . import mappings

ImportError: cannot import name core
```

```
$ conda list
# packages in environment at /Users/kayhan/anaconda:
#
_license                  1.1                      py27_0
abstract-rendering        0.5.1               np110py27_0
accelerate                1.12.0              np19py27_p0
accelerate_cudalib        2.0                           0
alabaster                 0.7.7                    py27_0
anaconda                  4.0.0               np110py27_0
anaconda-client           1.4.0                    py27_0
anaconda-navigator        1.1.0                    py27_0
appnope                   0.1.0                    py27_0
appscript                 1.0.1                    py27_0
argcomplete               1.0.0                    py27_1
astropy                   1.1.2               np110py27_0
autograd                  1.1.5                     <pip>
autoregressive            0.1.0                     <pip>
babel                     2.2.0                    py27_0
backports                 1.0                      py27_0
backports_abc             0.4                      py27_0
bcolz                     1.0.0                    py27_0
beautiful-soup            4.3.2                    py27_0
beautifulsoup4            4.4.1                    py27_0
binstar                   0.11.0                   py27_0
bitarray                  0.8.1                    py27_0
blaze                     0.9.1                    py27_0
blaze-core                0.9.0                    py27_0
blz                       removed                       0
bokeh                     0.11.1                   py27_0
boto                      2.39.0                   py27_0
bottleneck                1.0.0               np110py27_0
cdecimal                  2.3                      py27_0
certifi                   14.05.14                 py27_0
cffi                      1.5.2                    py27_0
chest                     0.2.3                    py27_0
brewer2mpl                1.4.1                    py27_0    Chroxvi
cloudpickle               0.1.1                    py27_0
clyent                    1.2.1                    py27_0
colorama                  0.3.7                    py27_0
conda                     4.1.11                   py27_0
conda-build               1.20.0                   py27_0
conda-env                 2.5.2                    py27_0
conda-manager             0.3.1                    py27_0
configobj                 5.0.6                    py27_0
cryptography              1.3                      py27_0
cudatoolkit               6.0                          p0
curl                      7.45.0                        0
cycler                    0.10.0                   py27_0
cython                    0.23.4                   py27_1
cytoolz                   0.7.5                    py27_0
dask                      0.8.1                    py27_0
datashape                 0.5.1                    py27_0
decorator                 4.0.10                    <pip>
decorator                 4.0.9                    py27_0
dill                      0.2.4                    py27_0
distribute                0.6.45                   py27_1
docutils                  0.12                     py27_0
flann                     1.8.5dev                      9    dougal
pyflann                   1.8.5dev                 py27_9    dougal
dynd-python               0.7.2                    py27_0
enum34                    1.1.2                    py27_0
et_xmlfile                1.0.1                    py27_0
fastcache                 1.0.2                    py27_0
flask                     0.10.1                   py27_1
flask-cors                2.1.2                    py27_0
freetype                  2.5.5                         0
funcsigs                  0.4                      py27_0
funcsigs                  1.0.2                     <pip>
future                    0.15.2                    <pip>
futures                   3.0.3                    py27_0
get_terminal_size         1.0.0                    py27_0
gevent                    1.1.0                    py27_0
gevent-websocket          0.9.5                    py27_1
glueviz                   0.7.1                    py27_0
GPy                       1.2.0                     <pip>
greenlet                  0.4.9                    py27_0
grin                      1.2.1                    py27_1
gslrandom                 0.1.1                     <pip>
h5py                      2.5.0               np110py27_4
hdf5                      1.8.15.1                      2
heapdict                  1.0.0                    py27_0
HIPS-Lib                  0.1                       <pip>
boost-vpython             1.55.0                        0    https://conda.binstar.org/mwcraig
fonttools                 2.4                  np19py27_0    https://conda.binstar.org/mwcraig
polygon2                  2.0.6                np19py27_0    https://conda.binstar.org/mwcraig
ttfquery                  1.0.5                    py27_0    https://conda.binstar.org/mwcraig
vpython                   6.11.0               np19py27_1    https://conda.binstar.org/mwcraig
idna                      2.0                      py27_0
iopro                     1.7.2               np19py27_p0
ipaddress                 1.0.14                   py27_0
ipykernel                 4.3.1                    py27_0
ipython                   4.1.2                    py27_1
ipython-notebook          4.0.4                    py27_0
ipython-qtconsole         4.0.1                    py27_0
ipython_genutils          0.1.0                    py27_0
ipywidgets                4.1.1                    py27_0
itsdangerous              0.24                     py27_0
jbig                      2.1                           0
jdcal                     1.2                      py27_0
jedi                      0.9.0                    py27_0
jinja2                    2.8                      py27_0
joblib                    0.9.4                    py27_0
jpeg                      8d                            1
jsonschema                2.4.0                    py27_0
jupyter                   1.0.0                    py27_2
jupyter_client            4.2.2                    py27_0
jupyter_console           4.1.1                    py27_0
jupyter_core              4.1.0                    py27_0
jupyterlab                0.1.2                     <pip>
launcher                  1.0.0                         3
libdynd                   0.7.2                         0
libgcc                    4.8.5                         1
libpng                    1.6.17                        0
libsodium                 0.4.5                         0
libtiff                   4.0.6                         1
libxml2                   2.9.2                         0
libxslt                   1.1.28                        2
llvmlite                  0.9.0                    py27_0
locket                    0.2.0                    py27_0
lxml                      3.6.0                    py27_0
markupsafe                0.23                     py27_0
matplotlib                1.5.1               np110py27_0
mistune                   0.7.2                    py27_1
mkl                       11.3.1                        0
mkl-rt                    11.1                         p0
mkl-service               1.1.2                    py27_0
mklfft                    2.0                 np19py27_p1
mock                      1.0.1                    py27_0
mock                      2.0.0                     <pip>
mpmath                    0.19                     py27_0
multipledispatch          0.4.8                    py27_0
nbconvert                 4.1.0                    py27_0
nbformat                  4.0.1                    py27_0
networkx                  1.11                     py27_0
nltk                      3.2                      py27_0
node-webkit               0.10.1                        0
nose                      1.3.7                    py27_0
notebook                  4.1.0                    py27_2
numba                     0.24.0              np110py27_0
numbapro                  0.19.0              np19py27_p0
numbapro_cudalib          0.2                           0
numexpr                   2.5                 np110py27_0
numpy                     1.10.4                   py27_0
numpy                     1.11.1                    <pip>
odo                       0.4.2                    py27_0
openpyxl                  2.3.2                    py27_0
openssl                   1.0.2h                        1
orange                    2.7.8                    py27_3
orange-app                1.0                      py27_0
pandas                    0.18.0              np110py27_0
paramz                    0.5.6                     <pip>
partd                     0.3.2                    py27_1
path.py                   8.1.2                    py27_1
pathlib2                  2.1.0                    py27_0
patsy                     0.4.0               np110py27_0
pbr                       1.9.1                     <pip>
pep8                      1.7.0                    py27_0
pexpect                   4.0.1                    py27_0
pickleshare               0.5                      py27_0
pillow                    3.1.1                    py27_0
pip                       8.1.2                     <pip>
pip                       8.1.2                    py27_0
ply                       3.8                      py27_0
prompt_toolkit            1.0.3                    py27_0
protobuf                  3.0.0b2                   <pip>
psutil                    4.1.0                    py27_1
ptyprocess                0.5                      py27_0
py                        1.4.31                   py27_0
pyasn1                    0.1.9                    py27_0
pyaudio                   0.2.7                    py27_0
pybasicbayes              0.2.1                     <pip>
pycosat                   0.6.1                    py27_0
pycparser                 2.14                     py27_0
pycrypto                  2.6.1                    py27_0
pycurl                    7.19.5.3                 py27_0
pyflakes                  1.1.0                    py27_0
pygments                  2.1.1                    py27_0
pyhsmm                    0.1.6                     <pip>
pylds                     0.0.2                     <pip>
pyopenssl                 0.15.1                   py27_2
pyparsing                 2.0.3                    py27_0
pypolyagamma              0.2                       <pip>
pyqt                      4.11.4                   py27_1
pystan                    2.8.0.0                  py27_0
pytables                  3.2.2               np110py27_1
pytest                    2.8.5                    py27_0
python                    2.7.11                        0
python-dateutil           2.5.1                    py27_0
python.app                1.2                      py27_4
pytz                      2016.2                   py27_0
pyyaml                    3.11                     py27_1
pyzmq                     15.2.0                   py27_0
qt                        4.8.7                         1
qtawesome                 0.3.2                    py27_0
qtconsole                 4.2.0                    py27_0
qtpy                      1.0                      py27_0
readline                  6.2                           2
redis                     2.6.9                         0
redis-py                  2.10.3                   py27_0
requests                  2.9.1                    py27_0
rope                      0.9.4                    py27_1
ruamel_yaml               0.11.7                   py27_0
runipy                    0.1.3                    py27_0
scikit-image              0.12.3              np110py27_0
scikit-learn              0.17.1              np110py27_0
scipy                     0.17.0              np110py27_0
scipy                     0.17.1                    <pip>
scipy                     0.18.0                    <pip>
setuptools                23.0.0                   py27_0
setuptools                24.0.2                    <pip>
simplegeneric             0.8.1                    py27_0
simpleitk                 0.9.1                    py27_0    simpleitk
singledispatch            3.4.0.3                  py27_0
sip                       4.16.9                   py27_0
six                       1.10.0                   py27_0
skl-groups                0.1.6                     <pip>
snakeviz                  0.4.0                    py27_0
snowballstemmer           1.2.1                    py27_0
sockjs-tornado            1.0.1                    py27_0
sphinx                    1.3.5                    py27_0
sphinx_rtd_theme          0.1.9                    py27_0
spyder                    2.3.8                    py27_1
spyder-app                2.3.8                    py27_0
sqlalchemy                1.0.12                   py27_0
sqlite                    3.13.0                        0
ssl_match_hostname        3.4.0.2                  py27_0
statsmodels               0.6.1               np110py27_0
sympy                     1.0                      py27_0
tensorflow                0.9.0                     <pip>
terminado                 0.5                      py27_1
tk                        8.5.18                        0
toolz                     0.7.4                    py27_0
tornado                   4.3                      py27_0
traitlets                 4.2.1                    py27_0
ujson                     1.33                     py27_0
unicodecsv                0.14.1                   py27_0
vpnotebook                0.1.3                    py27_0    vpython
wcwidth                   0.1.7                    py27_0
werkzeug                  0.11.4                   py27_0
wheel                     0.29.0                   py27_0
wxpython                  3.0                      py27_0
xlrd                      0.9.4                    py27_0
xlsxwriter                0.8.4                    py27_0
xlwings                   0.7.0                    py27_0
xlwt                      1.0.0                    py27_0
xz                        5.0.5                         1
yaml                      0.1.6                         0
zeromq                    4.1.3                         0
zlib                      1.2.8                         3
```

**Comment by mzwiessele** [comment]
Did you switch off the plotting library? It is definitely a matplotlib problem with the backend. Try import matplotlib before importing GPy, or set the plotting library off in the GPy user config file. ~/.config/gpy/user.cfg (or similar, keep forgetting the exact path).

**Comment by kayhan-batmanghelich** [comment]
@mzwiessele 

There is still problem. This is what I have:

```
import matplotlib
import GPy 
```

this is what i get:

```
/Users/kayhan/anaconda/lib/python2.7/site-packages/matplotlib/font_manager.py:273: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.
  warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')
An exception has occurred, use %tb to see the full traceback.

SystemExit: This program needs access to the screen.
Please run with a Framework build of python, and only when you are
logged in on the main display of your Mac.
To exit: use 'exit', 'quit', or Ctrl-D.
```

this is my config file:

```
$ cat ~/.config/gpy/user.cfg
# For user specific changes edit $HOME/.config/GPy/user.cfg

# [parallel]
# Enable openmp support. This speeds up some computations, depending on the number
# of cores available. Setting up a compiler with openmp support can be difficult on
# some platforms, hence by default it is off.
# openmp = False # True

# [datasets]
# location for the local data cache
# dir=$HOME/tmp/GPy-datasets/ # Any other path you choose

# [cython]
# working = True # False


# [plotting]
# library = none # matplotlib # plotly, none
```

**Comment by ktpolanski** [comment]
Just a quick update that it's all fine on my end now. I ran into that bizarre _slice_X() problem that was mentioned here in another issue, but installing everything from scratch from pip fixed it (somehow, even though I was already pipping in all the dependencies that this was relating to, i.e. paramz and that decorator thing... don't ask). Thanks for pushing the update out and for the mild thread hijack.
